schema.json

```diff
@@ -150,7 +150,28 @@
           "type": "string"
         }
       },
-      "required": ["column", "down", "name", "table", "type", "up"],
+      "required": ["table", "column"],
+      "oneOf": [
+        {
+          "required": ["name"],
+          "not": {
+            "required": ["up", "down"]
+          }
```

Comment:
I think this is not needed

Reply:
If this is removed then the following invalid `json` validates:

```json
{
  "table": "my_table",
  "column": "my_column",
  "name": "foo",
  "up": "foo",
  "down": "foo",
}
```

`up` and `down` are invalid when `name` is specified.

Reply:
But it shouldn't be `not.required`, the property should not be defined outside but inside the other part, right?

Reply:
I'm not sure what you mean here. Could you post what you think this `oneOf` section of the spec should look like?

Reply:
Sure, give me 30 min to finish a couple of things and I'll make a suggestion comment